functions/index.js

```diff
@@ -11,53 +11,454 @@ admin.initializeApp({
 // ==========================================================
 
 exports.signup = functions.https.onRequest((req, res) => {
```

Comment:
Can we restructure this an import the functions from other files this file is gonna get to large and is gonna become difficult to deal with in the future. 
https://stackoverflow.com/questions/43486278/how-do-i-structure-cloud-functions-for-firebase-to-deploy-multiple-functions-fro